### PR TITLE
Updates outdated Docs site URLs

### PR DESCRIPTION
**Pull request by ember-stevens**
**Requirements**

- [ n/a ] I have added test coverage for new or changed functionality
- [ X ] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests)
- [ n/a ] I have validated my changes against all supported platform versions

**Related issues**

N/A

**Describe the solution you've provided**

Updated old URLs linking to the old docs site to current URLs for the current docs site.

**Describe alternatives you've considered**

N/A

**Additional context**

Some docs site URLs changed in 2020 when we changed hosting platforms and there have been IA changes since.

### CODE REVIEW

File: src/LaunchDarkly.XamarinSdk/ILdClient.cs [hunk 1]
```diff
@@ -240,7 +240,7 @@ public interface ILdClient : IDisposable
         /// As of this version’s release date, the LaunchDarkly service does not support the <c>metricValue</c>
         /// parameter. As a result, calling this overload of <c>Track</c> will not yet produce any different
         /// behavior from calling <see cref="Track(String, LdValue)"/> without a <c>metricValue</c>.
```
Comment (eli-darkly) [inline src/LaunchDarkly.XamarinSdk/ILdClient.cs:242]: Whoops - this whole bit (lines 240-242) is obsolete and should be deleted.

Comment (ember-stevens) [reply src/LaunchDarkly.XamarinSdk/ILdClient.cs:242]:
```suggestion
```

File: src/LaunchDarkly.XamarinSdk/ILdClient.cs [hunk 2]
```diff
@@ -237,10 +237,7 @@ public interface ILdClient : IDisposable
         /// numeric metric value.
         /// </summary>
         /// <remarks>
-        /// As of this version’s release date, the LaunchDarkly service does not support the <c>metricValue</c>
-        /// parameter. As a result, calling this overload of <c>Track</c> will not yet produce any different
-        /// behavior from calling <see cref="Track(String, LdValue)"/> without a <c>metricValue</c>.
-        /// Refer to the <a href="https://docs.launchdarkly.com/docs/xamarin-sdk-reference#section-track">SDK reference guide</a>
+        /// Refer to the <a href="https://docs.launchdarkly.com/sdk/features/events#xamarin">SDK reference guide</a>
         /// for the latest status.
```
Comment (eli-darkly) [inline src/LaunchDarkly.XamarinSdk/ILdClient.cs:241]: Sorry, I should have included lines 240-241 in the obsolete section to be deleted. The "refer to ... for the latest status" part doesn't make sense now - it was talking about the "status" of the feature not having been implemented yet.

Comment (ember-stevens) [reply src/LaunchDarkly.XamarinSdk/ILdClient.cs:241]:
```suggestion
```

**Review by eli-darkly — APPROVED** [review]

**Pull request merged**